core/src/main/scala/kanaloa/dispatcher/Dispatcher.scala

```diff
@@ -172,27 +179,25 @@ case class PushingDispatcher(
 }
 
 object PushingDispatcher {
-
   def props[T: BackendAdaptor](
     name:       String,
     backend:    T,
     rootConfig: Config = ConfigFactory.load()
-  )(resultChecker: ResultChecker)(implicit system: ActorSystem) = {
-    val (settings, reporter) = Dispatcher.readConfig(name, rootConfig)
-    val metricsCollector = MetricsCollector(reporter, settings.performanceSamplerSettings)
+  )(resultChecker: ResultChecker)(implicit statsDClient: Option[StatsDClient]) = {
```

Comment:
But, we are still requiring that parameter be passed in, and the user now has to be mindful of an implicit parameter, in addition to the rest of the parameters.   I don't think this gains us any tangible benefit, and only makes the parameters a bit more confusing(in my opinion).  If there is concern about the arity, then why not have this object take a case class as a parameter instead?